### ISSUE TITLE
cli-image-create.test.js crashes if image creation isn't work

### ISSUE THREAD

**Issue by trentm** [joyent/node-triton#221]
The crash means that subsequent test files aren't run as part of `make test`. That sucks.
Instead it should just report test failure and continue on.

**Comment by trentm** [comment]
https://cr.joyent.us/#/c/2351/

**Comment by trentm** [comment]
After this change a test run with failing image creation looks like this:

```
# triton image ...
# Test config:
# - profileName: "nightly-1"
# - allowWriteActions: true
# - skipAffinityTests: true
# - skipKvmTests: true
# - profile: {"account":"trent.mick","insecure":true,"keyId":"SHA256:2XCGHtsnTro2MGqoVQHgXSS/ai12yQHrGFrN12YUPpA","url":"https://172.26.1.8","name":"nightly-1"}
#   setup: rm existing origin inst nodetritontest-images-danger0.local-origin
ok 134 ran "triton inst rm -w b18206b5-ae03-43b1-8d48-80409fadca8d", err=null
ok 135 empty stderr
ok 136 deleted inst b18206b5-ae03-43b1-8d48-80409fadca8d
#   setup: rm existing derived inst nodetritontest-images-danger0.local-derived
ok 137 no pre-existing inst nodetritontest-images-danger0.local-derived
#   setup: rm existing img nodetritontest-images-danger0.local@1.0.0
ok 138 no pre-existing img nodetritontest-images-danger0.local@1.0.0
#   setup: find origin image
ok 139 ran "triton img ls -j", err=null
ok 140 empty stderr
ok 141 latest available base/minimal image: 1f32508c-e6e9-11e6-bc05-8fea9e979940 (base-64-lts@16.4.1)
ok 142 getTestImg
#   setup: find test package
ok 143 ran "triton pkg ls -j", err=null
ok 144 empty stderr
ok 145 smallest (RAM) available package: 1151ec8f-1326-4338-d3b8-d5801d88a18c (sample-128M)
ok 146 getTestPkg
#   setup: triton create ... -n nodetritontest-images-danger0.local-origin
ok 147 ran "triton create -wj -n nodetritontest-images-danger0.local-origin -m user-script=touch /nodetritontest-was-here.txt 1f32508c-e6e9-11e6-bc05-8fea9e979940 1151ec8f-1326-4338-d3b8-d5801d88a18c", err=null
ok 148 empty stderr
ok 149 originInst.id: bb44e2f4-7518-c4c1-bf7d-dc502e4c8757
ok 150 originInst is running
#   triton image create ...
not ok 151 ran "triton image create -j -w -t foo=bar bb44e2f4-7518-c4c1-bf7d-dc502e4c8757 nodetritontest-images-danger0.local 1.0.0", err=VError: exec error: command: /Users/trentm/opt/node-4.8.0/bin/node,/Users/trentm/joy/node-triton/bin/triton,image,create,-j,-w,-t,foo=bar,bb44e2f4-7518-c4c1-bf7d-dc502e4c8757,nodetritontest-images-danger0.local,1.0.0 exit status: 1 stdout: {"id":"51b4c836-9e29-4653-a125-ae968651945f","name":"nodetritontest-images-danger0.local","version":"1.0.0","requirements":{},"tags":{"foo":"bar"},"state":"creating"} stderr: triton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found: Command failed: /Users/trentm/opt/node-4.8.0/bin/node /Users/trentm/joy/node-triton/bin/triton image create -j -w -t foo=bar bb44e2f4-7518-c4c1-bf7d-dc502e4c8757 nodetritontest-images-danger0.local 1.0.0 triton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found
  ---
    operator: error
    expected: |-
      undefined
    actual: |-
      { code: 1, jse_cause: { [Error: Command failed: /Users/trentm/opt/node-4.8.0/bin/node /Users/trentm/joy/node-triton/bin/triton image create -j -w -t foo=bar bb44e2f4-7518-c4c1-bf7d-dc502e4c8757 nodetritontest-images-danger0.local 1.0.0
triton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found
] killed: false, code: 1, signal: null, cmd: '/Users/trentm/opt/node-4.8.0/bin/node /Users/trentm/joy/node-triton/bin/triton image create -j -w -t foo=bar bb44e2f4-7518-c4c1-bf7d-dc502e4c8757 nodetritontest-images-danger0.local 1.0.0' }, jse_shortmsg: 'exec error:\n\tcommand: /Users/trentm/opt/node-4.8.0/bin/node,/Users/trentm/joy/node-triton/bin/triton,image,create,-j,-w,-t,foo=bar,bb44e2f4-7518-c4c1-bf7d-dc502e4c8757,nodetritontest-images-danger0.local,1.0.0\n\texit status: 1\n\tstdout:\n{"id":"51b4c836-9e29-4653-a125-ae968651945f","name":"nodetritontest-images-danger0.local","version":"1.0.0","requirements":{},"tags":{"foo":"bar"},"state":"creating"}\n\tstderr:\ntriton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found', jse_summary: 'exec error:\n\tcommand: /Users/trentm/opt/node-4.8.0/bin/node,/Users/trentm/joy/node-triton/bin/triton,image,create,-j,-w,-t,foo=bar,bb44e2f4-7518-c4c1-bf7d-dc502e4c8757,nodetritontest-images-danger0.local,1.0.0\n\texit status: 1\n\tstdout:\n{"id":"51b4c836-9e29-4653-a125-ae968651945f","name":"nodetritontest-images-danger0.local","version":"1.0.0","requirements":{},"tags":{"foo":"bar"},"state":"creating"}\n\tstderr:\ntriton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found: Command failed: /Users/trentm/opt/node-4.8.0/bin/node /Users/trentm/joy/node-triton/bin/triton image create -j -w -t foo=bar bb44e2f4-7518-c4c1-bf7d-dc502e4c8757 nodetritontest-images-danger0.local 1.0.0\ntriton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found\n', message: 'exec error:\n\tcommand: /Users/trentm/opt/node-4.8.0/bin/node,/Users/trentm/joy/node-triton/bin/triton,image,create,-j,-w,-t,foo=bar,bb44e2f4-7518-c4c1-bf7d-dc502e4c8757,nodetritontest-images-danger0.local,1.0.0\n\texit status: 1\n\tstdout:\n{"id":"51b4c836-9e29-4653-a125-ae968651945f","name":"nodetritontest-images-danger0.local","version":"1.0.0","requirements":{},"tags":{"foo":"bar"},"state":"creating"}\n\tstderr:\ntriton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found: Command failed: /Users/trentm/opt/node-4.8.0/bin/node /Users/trentm/joy/node-triton/bin/triton image create -j -w -t foo=bar bb44e2f4-7518-c4c1-bf7d-dc502e4c8757 nodetritontest-images-danger0.local 1.0.0\ntriton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found\n' }
    at: maybeClose (internal/child_process.js:854:16)
    stack: |-
      VError: exec error:
      	command: /Users/trentm/opt/node-4.8.0/bin/node,/Users/trentm/joy/node-triton/bin/triton,image,create,-j,-w,-t,foo=bar,bb44e2f4-7518-c4c1-bf7d-dc502e4c8757,nodetritontest-images-danger0.local,1.0.0
      	exit status: 1
      	stdout:
      {"id":"51b4c836-9e29-4653-a125-ae968651945f","name":"nodetritontest-images-danger0.local","version":"1.0.0","requirements":{},"tags":{"foo":"bar"},"state":"creating"}
      	stderr:
      triton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found: Command failed: /Users/trentm/opt/node-4.8.0/bin/node /Users/trentm/joy/node-triton/bin/triton image create -j -w -t foo=bar bb44e2f4-7518-c4c1-bf7d-dc502e4c8757 nodetritontest-images-danger0.local 1.0.0
      triton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found

          at /Users/trentm/joy/node-triton/test/lib/testcommon.js:48:27
          at ChildProcess.exithandler (child_process.js:209:5)
          at emitTwo (events.js:87:13)
          at ChildProcess.emit (events.js:172:7)
          at maybeClose (internal/child_process.js:854:16)
          at Process.ChildProcess._handle.onexit (internal/child_process.js:222:5)
  ...
not ok 152 empty stderr
  ---
    operator: equal
    expected: |-
      ''
    actual: |-
      'triton image create: error (ResourceNotFound): image 51b4c836-9e29-4653-a125-ae968651945f not found\n'
    at: maybeClose (internal/child_process.js:854:16)
  ...
#   triton create ... -n nodetritontest-images-danger0.local-derived
not ok 153 have an img to test
  ---
    operator: ok
    expected: true
    actual:   undefined
    at: Test.<anonymous> (/Users/trentm/joy/node-triton/test/integration/cli-image-create.test.js:184:11)
  ...
#   cleanup: triton rm
#   cleanup: triton image rm
...
```
